File: solutions/7kyu/Palindrome_chain_length/palindrome_chain_length.rb
def palindrome_chain_length(number)
  count(number, 0)
end

def count(number, iterator)
  p number
  number == number.to_s.reverse.to_i ? iterator : count(number.to_s.reverse.to_i + number, iterator += 1)
end

p palindrome_chain_length(87) #= 4

File: solutions/7kyu/Palindrome_chain_length/palindrome_chain_length.js
let palindromeChainLength = function pow(number) {
  return number == number.toString().split("").reverse().join("") ? 0 : 1 + palindromeChainLength(number + parseInt(number.toString().split("").reverse().join("")));
}

console.log(palindromeChainLength(87)) //#= 4
